src/Stratis.Bitcoin.Features.Consensus/PowConsensusValidator.cs

```diff
@@ -657,7 +657,7 @@ public virtual void CheckTransaction(Transaction transaction)
             }
 
             // Size limits (this doesn't take the witness into account, as that hasn't been checked for malleability).
-            if (this.GetSize(transaction, NetworkOptions.None) > this.ConsensusOptions.MaxBlockBaseSize)
+            if (this.GetSize(transaction, transaction.GetNetworkOptions() & ~NetworkOptions.Witness) > this.ConsensusOptions.MaxBlockBaseSize)
```

Comment:
How can we guarantee in POS this will be set correctly?

Reply:
The POS flag would have to be set correctly for the transaction object.